// @algolia/autocomplete-js/dist/esm/getPanelPlacementStyle.js
export function getPanelPlacementStyle(_ref) {
  var panelPlacement = _ref.panelPlacement,
    container = _ref.container,
    form = _ref.form,
    environment = _ref.environment;
  var containerRect = container.getBoundingClientRect();
  var offsetParent = container.offsetParent || environment.document.documentElement;
  var top = container.offsetTop + containerRect.height;
  switch (panelPlacement) {
    case 'start':
      {
        return {
          top: top,
          left: container.offsetLeft
        };
      }
    case 'end':
      {
        return {
          top: top,
          right: offsetParent.clientWidth - (container.offsetLeft + containerRect.width)
        };
      }
    case 'full-width':
      {
        return {
          top: top,
          left: 0,
          right: 0,
          width: 'unset',
          maxWidth: 'unset'
        };
      }
    case 'input-wrapper-width':
      {
        var formRect = form.getBoundingClientRect();
        return {
          top: top,
          left: form.offsetLeft,
          right: offsetParent.clientWidth - (form.offsetLeft + formRect.width),
          width: 'unset',
          maxWidth: 'unset'
        };
      }
    default:
      {
        throw new Error("[Autocomplete] The `panelPlacement` value ".concat(JSON.stringify(panelPlacement), " is not valid."));
      }
  }
}
